packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -86,25 +85,20 @@ macro_rules! impl_tx_trait {
         impl TransactionBuilder for $ty {
             type TxType = $tx_ty;
             fn build(self) -> Result<$tx_ty> {
-                let uses_predicates = self.is_using_predicates();
-                let base_offset = if uses_predicates {
+                let is_using_predicates = self.is_using_predicates();
+                let base_offset = if is_using_predicates {
                     self.base_offset()
                 } else {
                     0
                 };
 
-                let network_info = self.network_info.clone();
-
                 let num_witnesses = self.num_witnesses()?;
-                let mut tx = self.resolve_fuel_tx(base_offset, num_witnesses)?;
-
-                tx.precompute(&network_info.chain_id())?;
-
-                if uses_predicates {
-                    estimate_predicates(&mut tx, &network_info)?;
-                };
+                let tx = self.resolve_fuel_tx(base_offset, num_witnesses)?;
 
-                Ok($tx_ty { tx })
+                Ok($tx_ty {
+                    tx,
+                    is_using_predicates,
```

Comment:
```suggestion
                    self.is_using_predicates(),
```

Reply:
Thanks for the suggestion. As the above is a sturct, I would have to write 
```rust
                Ok($tx_ty {
                    tx,
                    is_using_predicates: self.is_using_predicates(),
                })
```
IMHO the old version is nicer.